ui/src/components/CloneDialog.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { useContext } from "react";
+import {
+  Button,
+  TextField,
+  Typography,
+  Grid,
+  Backdrop,
+  CircularProgress,
+} from "@mui/material";
+import Dialog from "@mui/material/Dialog";
+import DialogActions from "@mui/material/DialogActions";
+import DialogContent from "@mui/material/DialogContent";
+import DialogContentText from "@mui/material/DialogContentText";
+import DialogTitle from "@mui/material/DialogTitle";
+import { createDockerDesktopClient } from "@docker/extension-api-client";
+
+import { MyContext } from "../index";
+
+const client = createDockerDesktopClient();
+
+function useDockerDesktopClient() {
+  return client;
+}
+
+export default function CloneDialog({ ...props }) {
+  console.log("CloneDialog component rendered.");
+  const ddClient = useDockerDesktopClient();
+  const context = useContext(MyContext);
+
+  const [volumeName, setVolumeName] = React.useState<string>(
+    `${context.store.volumeName}-cloned`
+  );
+  const [actionInProgress, setActionInProgress] =
+    React.useState<boolean>(false);
+
+  const cloneVolume = async () => {
+    setActionInProgress(true);
+
+    try {
+      // TODO: check if destination volume already exists
+      console.log(`Creating destination volume ${volumeName}...`);
+      const createVolumeOutput = await ddClient.docker.cli.exec("volume", [
+        "create",
+        volumeName,
+      ]);
+      if (createVolumeOutput.stderr !== "") {
+        ddClient.desktopUI.toast.error(createVolumeOutput.stderr);
+        return;
+      }
+      console.log(createVolumeOutput);
+
+      console.log(
+        `Copying data from source volume ${context.store.volumeName} to destination volume ${volumeName}...`
+      );
```

Comment:
nit: remove

Reply:
if it is useful, maybe you should add this message in the UI? See https://mui.com/material-ui/react-snackbar/#customized-snackbars